src/Microsoft.DotNet.XHarness.CLI/CommandArguments/WASI/Arguments/LibraryArgument.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.DotNet.XHarness.CLI.CommandArguments.Wasi;
+
+internal class LibraryArgument : RequiredStringArgument
+{
+    public LibraryArgument(string defaultValue)
```

Comment:
This shouldn't be a separate argument. Instead, it can passed by the user as part of the arguments.
One reason is that the main dll name is a something specific to the .wasm file that we are running, and it might not be actually needed in some cases. For example, when we bundle all the assemblies, and dependencies in a single `foo.wasm` file then it has the name of the main assembly embedded, and does not need this as an argument.